### PR TITLE
Streamlines/Fixes Piracy Spawner Checks - They Piss Off When Paid

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Fixes #74211.

I think at some point this worked, but there were just too many checks on abstract concepts and it seems that "aborting" the callback wasn't really working? Anyways everything just felt really awk to me with how `message_override` was working, so let's store a variable on `chosen_gang` since we're already reliant on that for all of our messages and stuff like that, and just slim down the number of weird checks that we do into just one simple "did we get paid"?

One consequence of this is that refusing to pay won't spawn pirates instantly, but the time-to-respond was already two minutes so I don't really see this as a major drawback.
## Why It's Good For The Game

I PAID THEM OFF NOW FUCK OFF
## Changelog
:cl:
fix: Pirates finally realized that they shouldn't risk life and limb attacking stations if they had already paid them off.
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
>it's not a feature, here's a good video on why https://www.youtube.com/watch?v=JzOuJvw1XhU

I think i understand what you meant, but just look at this from another side.

If you're planning to rob someone completelly, it'll be worth a chance to just ask them to give you at least something from their stash, and if they believe you, you're still robbing them so it's just less affort for you.

**Comment by san7890** [comment]
yeah but you want to LIVE. think of this less like a video game and more like real life: easy money is easy money

**Comment by Helg2** [comment]
> yeah but you want to LIVE. think of this less like a video game and more like real life: easy money is easy money

I doubt pirate psykers want to live like that xD

File: code/modules/antagonists/pirate/pirate_event.dm [hunk 1]
```diff
@@ -34,24 +34,24 @@
...
-	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(spawn_pirates), threat, chosen_gang, FALSE), RESPONSE_MAX_TIME)
+	addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(spawn_pirates), threat, chosen_gang), RESPONSE_MAX_TIME)
 	SScommunications.send_message(threat, unique = TRUE)
 
 /proc/pirates_answered(datum/comm_message/threat, datum/pirate_gang/chosen_gang, payoff, initial_send_time)
 	if(world.time > initial_send_time + RESPONSE_MAX_TIME)
-		priority_announce(chosen_gang.response_too_late ,sender_override = chosen_gang.ship_name)
+		priority_announce(chosen_gang.response_too_late, sender_override = chosen_gang.ship_name)
 		return
 	if(threat?.answered)
```
Comment (Jacquerel) [inline code/modules/antagonists/pirate/pirate_event.dm:44]: Could we early return here?

**Review by Jacquerel — CHANGES_REQUESTED** [review]

**Pull request merged**